crates/sui-bridge-indexer/src/eth_worker.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::postgres_manager::{write, PgPool};
+use crate::{BridgeDataSource, TokenTransfer, TokenTransferData, TokenTransferStatus};
+use anyhow::Result;
+use ethers::providers::Provider;
+use ethers::providers::{Http, Middleware};
+use ethers::types::Address as EthAddress;
+use std::sync::Arc;
+use sui_bridge::abi::{EthBridgeEvent, EthSuiBridgeEvents};
+use sui_bridge::types::EthLog;
+use tracing::info;
+
+pub async fn process_eth_events(
+    mut eth_events_rx: mysten_metrics::metered_channel::Receiver<(EthAddress, u64, Vec<EthLog>)>,
+    provider: Arc<Provider<Http>>,
+    pool: &PgPool,
+    finalized: bool,
+) -> Result<()> {
+    while let Some((_, _, logs)) = eth_events_rx.recv().await {
+        for log in logs.iter() {
+            let eth_bridge_event = EthBridgeEvent::try_from_eth_log(log);
+            if eth_bridge_event.is_none() {
+                continue;
+            }
+            let bridge_event = eth_bridge_event.unwrap();
+            let block_number = log.block_number;
+            let block = provider.get_block(log.block_number).await.unwrap().unwrap();
```

Comment:
should we log an `error!` instead of just `unwrap`? something like `unwrap_or_else` with an error and a `panic!`.
I am not sure if a `panic!` would amount to the same though

Reply:
yes let's not unwrap here. The eth node may send back an malformed response when there is a bug, we don't want indexer to crash because of that.

let's do a full audit of all unwrap/expect for indexer